packages/sdk/src/protocols/filter.ts

```diff
@@ -57,28 +62,37 @@ export class SubscriptionManager {
   async subscribe<T extends IDecodedMessage>(
     decoders: IDecoder<T> | IDecoder<T>[],
     callback: Callback<T>
-  ): Promise<void> {
+  ): Promise<SDKProtocolResult> {
     const decodersArray = Array.isArray(decoders) ? decoders : [decoders];
 
     // check that all decoders are configured for the same pubsub topic as this subscription
-    decodersArray.forEach((decoder) => {
+    for (const decoder of decodersArray) {
       if (decoder.pubsubTopic !== this.pubsubTopic) {
-        throw new Error(
-          `Pubsub topic not configured: decoder is configured for pubsub topic ${decoder.pubsubTopic} but this subscription is for pubsub topic ${this.pubsubTopic}. Please create a new Subscription for the different pubsub topic.`
-        );
+        return {
+          failures: [
+            {
+              error: ProtocolError.TOPIC_DECODER_MISMATCH
```

Comment:
When passing multiple decoders, this does not tell the user which decoder is mismatched.

Reply:
It's not about _which_ decoder, but about a mismatch in general.

Example:
`DecoderA`: `/abc/xyz`
`DecoderB`: `/qwe/rty`

So here, it's not about which decoder doesn't match the other, but the binary result that either they **all** match, or they don't.

Wdyt? @adklempner

Reply:
The comparison is made between the decoder's pubsub topic and the subscription manager's pubsub topic. So you can have a case where all the decoders match each other but do not match the pubsub topic set in the subscription manager. This can be quickly detected if we log both pubsub topics.

Reply:
I think if we go with changes described in this issue then it won't be as much of a problem: https://github.com/waku-org/js-waku/issues/1981

Reply:
i like this change